test/integration-test.sh

```diff
@@ -45,74 +46,50 @@ declare api6="${6}"
-  # no timeout set since the test execution environment should cancel the test if it takes too long
-  local cmd="curl -m ${step_time} --connect-timeout ${step_time} -s -H X-Auth-Token:${api_token} -H Content-Type:application/json --url ${endpoint}/api/v2/messages -o /dev/null -w %{http_code} -d "
-
-  # if no end time was given we need to calculate it once
-  if [ ${end_time_ns} -eq 0 ]; then
-    now=$(node -e "console.log(process.hrtime.bigint().toString());")
-    # need to calculate in nanoseconds
-    ((end_time_ns=now+wait_time*1000000000))
-  fi
+# $1 = node api address (origin)
+# validate that node is funded
+validate_node_balance_gt0() {
+  local balance eth_balance hopr_balance
+  local endpoint=${1:-localhost:3001}
 
-  local path=$(echo ${peers} | tr -d '\n' | jq -R -s 'split(" ")')
-  local message='{"body":"'${msg}'","path":'${path}',"recipient":"'${recipient}'"}'
-  result=$(${cmd} "${message}")
+  balance=$(get_balances ${1})
+  eth_balance=$(echo ${balance} | jq -r ".native")
+  hopr_balance=$(echo ${balance} | jq -r ".hopr")
 
-  # we fail if the HTTP status code is anything but 204
-  if [ "${result}" = "204" ]; then
-    echo "${result}"
+  if [[ "$eth_balance" != "0" && "$hopr_balance" != "0" ]]; then
+    log "$1 is funded"
   else
-    now=$(node -e "console.log(process.hrtime.bigint().toString());")
-    if [ ${end_time_ns} -lt ${now} ]; then
-      log "${RED}send_message (${cmd} \"${message}\") FAILED, received: ${result}${NOFORMAT}"
-      exit 1
-    else
-      log "${YELLOW}send_message (${cmd} \"${message}\") FAILED, retrying in ${step_time} seconds${NOFORMAT}"
-      sleep ${step_time}
-      send_message "${endpoint}" "${recipient}" "${msg}" "${peers}" "${wait_time}" "${step_time}" "${end_time_ns}"
-    fi
+    log "⛔️ $1 Node has an invalid balance: $eth_balance, $hopr_balance"
+    log "$balance"
+    exit 1
   fi
 }
 
-# $1 = endpoint
-# $2 = Hopr command
-# $3 = OPTIONAL: positive assertion message
-# $4 = OPTIONAL: maximum wait time in seconds during which we busy try
-# afterwards we fail, defaults to 0
-# $4 = OPTIONAL: step time between retries in seconds, defaults to 25 seconds
-# (8 blocks with 1-3 s/block in ganache)
-# $5 = OPTIONAL: end time for busy wait in nanoseconds since epoch, has higher
-# priority than wait time, defaults to 0
-run_command(){
+# Run API endpoint and assert response or status code
+# $1 = source api url (http://localhost:13001)
+# $2 = api_endpoint (/channels)
+# $3 = rest method for cURL (GET,POST...)
+# $4 = request body as json string
+# $5 = OPTIONAL: positive assertion message
+# $6 = OPTIONAL: maximum wait time in seconds during which we busy try afterwards we fail, defaults to 0
+# $7 = OPTIONAL: step time between retries in seconds, defaults to 25 seconds (8 blocks with 1-3 s/block in ganache)
+# $8 = OPTIONAL: end time for busy wait in nanoseconds since epoch, has higher priority than wait time, defaults to 0
+# $9 = OPTIONAL: should assert status code
+run_api(){
```

Comment:
The name of the function is misleading. The function makes an API call and doesn't run the API.

Reply:
renamed to `call_api`